### PR TITLE
Custom point type

### LINKED ISSUE #200: Define custom pcl point_type to have beam origin
Current implementation stores id in `intensity` field.
The type should have `beam_origin_id`.

### PR DESCRIPTION
**Pull request by at-wat**
fix #200

### CODE REVIEW

**Comment by at-wat** [comment]
## [#530] FAILED on indigo

<details><summary>Test failed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests, 0 errors, 1 failures, 0 skipped
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 36 tests, 0 errors, 1 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#531] FAILED on indigo

```catkin_make``` failed

**Comment by at-wat** [comment]
backport https://github.com/PointCloudLibrary/pcl/commit/5e19c141949ab61fd936a1bacd895e3ed4183a4b#diff-376b015c9fde4a9d71e86d2d49197523 or drop indigo support

**Comment by at-wat** [comment]
## [#565] FAILED on indigo

```catkin_make``` failed

**Comment by at-wat** [comment]
## [#569] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#570] FAILED on melodic

<details><summary>Test failed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests, 0 errors, 1 failures, 0 skipped
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 1 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#571] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#572] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#572] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#573] PASSED on kinetic

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#574] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: src/mcl_3dl.cpp [hunk 1]
```diff
@@ -109,42 +112,49 @@ class MCL3dlNode
   void cbMapcloud(const sensor_msgs::PointCloud2::ConstPtr& msg)
   {
     ROS_INFO("map received");
-    pcl::PointCloud<pcl::PointXYZI> pc_tmp;
-    pcl::fromROSMsg(*msg, pc_tmp);
-
-    if (pc_tmp.points.size() == 0)
+    pcl::PointCloud<pcl::PointXYZI>::Ptr pc_tmp_raw(new pcl::PointCloud<pcl::PointXYZI>);
+    pcl::fromROSMsg(*msg, *pc_tmp_raw);
```
Comment (at-wat) [inline src/mcl_3dl.cpp]: switch by input point type to remove warnings (`Failed to find match for field 'intensity'.`).

**Comment by at-wat** [comment]
## [#575] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#576] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
## [#577] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

**Comment by at-wat** [comment]
@DaikiMaekawa please take a look

File: .travis.yml [hunk 2]
```diff
@@ -19,9 +19,9 @@ env:
     - DATASET_REF_FILE=short_test_ref.topic
     - secure: "Nk9LAc9NWbnf5urjmTe1nQHdzdbcKV/gYVJB2EBHiQafsn28IQ1f4BBd8tJ7+iFxnkM4I2kGtWTCvfdM91nITz0dj+MN/wbKQ0JsC1KFSi2j0ZRfnfqdQV7s3gfbGJWCQyWR3Z1RLa5FyCTNIb1Ov0fGe3/bYRiuI2mY5cKhhTuEu1grS2gFEjPiSMn3huuqGRpHYzA+a9CNfSHa9msz2AwJggsbbXhGlEz/WXZC0Irc/cLjiTtaqe82cMqxjlzmwVubZZNGHOoRmV7X7aDQmywLgySIx0eGwioz61Tz/puQ6VaZHR4eOUsuF8Jfv3HttwK56cWwQI7XraQokrMJ+fZM4qsg5v0miOeXnPSSZ3baKvAyGBcEQKRUUxiHJ8T5oYJvcRVTQWGM2WSDDlMVAG7jevyinZ9VKYEavNp5P3g7qbi6k2yX0nZ7MVYFJ9pSdowSkuh6DqCE7SnzpTYEHmgdbw5K6fPSzAoCxeKWEN2+3p/iHLT4Iw/iiEbg+FT42eXjHwhfjmNogfko8bd0phjo7R7xxM7q1pNiK4PtZpBdh/RSUgYtTNDWgDDLESgF3tTLBIH0th8b2nUFxW3NIdsY+hOrFkVuhoAd48Enepu6Ft9mA5c7iQCS0b534ah+KpTgd0D0PAQXoF9fp/qP7ri0Zr/MFSvGM4miR2HRvb8="
   matrix:
-    - ROS_DISTRO_TARGET=melodic
-    - ROS_DISTRO_TARGET=kinetic
-    - ROS_DISTRO_TARGET=indigo
+    - ROS_DISTRO_TARGET=melodic COVERAGE_TEST=false
+    - ROS_DISTRO_TARGET=kinetic COVERAGE_TEST=false
```
Comment (DaikiMaekawa) [inline .travis.yml]: Why did you disable the coverage tests for these?

Comment (at-wat) [reply .travis.yml]: Large amount of the code backported in this PR is only used on indigo. On the other distros, coverage is evaluated very small.

Comment (at-wat) [reply .travis.yml]: codecov seems merging reports from multiple jobs.
https://docs.codecov.io/docs/merging-reports
> Codecov supports multiple languages in the same repository. You may upload reports for one or more languages, and Codecov will merge the reports automatically while maintaining the original upload context.
>
> Codecov does not override report data for multiple uploads. We always merge the data. Simply upload all three reports at once, or separately.

this change might be able to be reverted without negative effects.

**Comment by at-wat** [comment]
## [#578] PASSED on indigo

<details><summary>All tests passed</summary>


```
build/test_results/mcl_3dl/gtest-test_chunked_kdtree.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_filter.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_nd.xml: 2 tests
build/test_results/mcl_3dl/gtest-test_pf.xml: 5 tests
build/test_results/mcl_3dl/gtest-test_point_cloud_random_sampler.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_point_types.xml: 1 tests
build/test_results/mcl_3dl/gtest-test_quat.xml: 4 tests
build/test_results/mcl_3dl/gtest-test_raycast.xml: 3 tests
build/test_results/mcl_3dl/gtest-test_vec3.xml: 5 tests
build/test_results/mcl_3dl/roslint-mcl_3dl.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_expansion_resetting_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_global_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_localization_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_mcl_3dl_compat_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_tf_rostest.xml: 1 tests
build/test_results/mcl_3dl/rostest-test_tests_transform_rostest.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_pose.xml: 1 tests
build/test_results/mcl_3dl/rosunit-compare_tf.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_expansion_resetting.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_global_localization.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_mcl_3dl_compat.xml: 1 tests
build/test_results/mcl_3dl/rosunit-test_transform_failure.xml: 1 tests
Summary: 38 tests, 0 errors, 0 failures, 0 skipped
```
</details>

File: .travis.yml (hunk 2 above)
Comment (at-wat) [reply .travis.yml]: coverage reports looks correctly merged

**Comment by at-wat** [comment]
@DaikiMaekawa PTAL

**Comment by DaikiMaekawa** [comment]
LGTM

**Pull request merged**